scripts/generate_example_ttl_files.py

```diff
@@ -0,0 +1,120 @@
+"""Script to generate all example .ttl files from example CSV inputs.
+
+Should be run after making any changes to template mapping,
+to update the example ttl files with the result of those mapping changes."""
+
+# standard library
+import collections
+import collections.abc
+import hashlib
+import inspect
+import itertools
+import pathlib
+import unittest.mock
+
+# third-party
+import rdflib
+
+# local
+import abis_mapping
+import abis_mapping.types.temporal
+import tests.templates.conftest
+
+# Global vars
+COUNTERS: collections.abc.Mapping[bytes, collections.abc.Iterator[int]] = collections.defaultdict(itertools.count)
+TTL_FILE_PATH: str = ""
+
+
+def main() -> None:
+    global TTL_FILE_PATH
+
+    # Gather the example files to be regenerated.
+    examples: list[tuple[str, pathlib.Path, pathlib.Path]] = [
+        (template_test.template_id, mapping_test.data, mapping_test.expected)
+        for template_test in tests.templates.conftest.TEST_CASES
+        for mapping_test in template_test.mapping_cases
+        if mapping_test.expected is not None
+    ]
+
+    # In order to produce consistent ttl output, and reduce the git diff
+    # each time the mapping is changed, this script overrides some behavior.
+
+    # 1. First override:
+    # Here we override the BNode id generation function to generate IDs based on the
+    # hash on the ttl file path being generated, and the
+    # current stack of functions that were called to reach the BNode() call.
+    # This is so that BNode IDs that are embedded in the ttl output,
+    # are consistent between runs of this script,
+    # and are consistent for a particular code path.
+
+    # Check rdflib.BNode.__new__ has signature we expect
+    b_node_signature = inspect.signature(rdflib.BNode.__new__)
+    expected_params = ["cls", "value", "_sn_gen", "_prefix"]
+    actual_params = [p.name for p in b_node_signature.parameters.values()]
+    defaults = rdflib.BNode.__new__.__defaults__
+    if actual_params != expected_params or defaults != (None, None, "N"):
+        raise RuntimeError("BNode.__new__ had unexpected signature, the following code might need to be updated")
+    # Override default _sn_gen function with our custom one.
+    rdflib.BNode.__new__.__defaults__ = (defaults[0], _custom_sn_gen, defaults[2])
+
+    # 2. second override:
+    # Mock "Date.today()" function to always give the same result
+    unittest.mock.patch.object(
+        abis_mapping.types.temporal.Date,
+        "today",
+        new=lambda: abis_mapping.types.temporal.Date(2020, 1, 1),
```

Comment:
please set to your birthday or another date of significance to you :stuck_out_tongue_winking_eye: